src/db/units.ts

```diff
@@ -92,22 +93,53 @@ export async function insertOrUpdateUnit(unit: Unit): Promise<void> {
   }
 }
 
-export async function unitGone(unitId: number): Promise<void> {
+export async function unitGone(unit: Pick<Unit, 'name'>): Promise<void> {
+  const { name } = unit
+
   const timestamp = new Date()
 
   await knex('units')
     .update({
       updatedAt: timestamp,
       goneAt: timestamp,
     })
-    .where({ unitId })
+    .where({ name })
+}
+
+export async function findUnit(name: string): Promise<Unit | undefined> {
+  const foundUnit = await knex('units')
+    .select('*')
+    .where({ name })
+    .leftJoin('positions', 'units.positionId', 'positions.positionId')
+    .first()
+
+  if (foundUnit) {
+    const { alt, country, lat, lon, name, typeName, unitId: id } = foundUnit
+    return {
+      id,
+      name,
+      coalition: coalitionFrom(country),
+      type: typeName,
+      position: {
+        lat,
+        lon,
+        alt,
+      },
+      // TODO: determine what to do with the below in the types
```

Comment:
I haven't put much thought into this, but we have two `Unit` types, intentionally. There is the "dcs unit" which comes from `dcs-grpc` and a "database unit" which is the unit represented in our database.

I think I'm leaning towards having a `GroundUnit` type instead of a `Unit` type, and `Unit` should be a union of the different unit types.
and then we'd just convert from dcs-grpc types to our internal specific unit types before we use anything in functions with a `groundUnitFrom` type fn